lib/dotcom_web/live/subway_alerts.ex

```diff
@@ -0,0 +1,159 @@
+defmodule DotcomWeb.Live.SubwayAlerts do
+  @moduledoc """
+  The subway alerts/status page, showing current subway status, planned work, and
+  other upcoming and relevant alerts.
+  """
+
+  use DotcomWeb, :live_view
+
+  import DotcomWeb.Components.PlannedDisruptions, only: [disruptions: 1]
+  import DotcomWeb.Components.TAlerts, only: [t_alerts: 1]
+  import DotcomWeb.Components.SystemStatus.SubwayStatus, only: [alerts_subway_status: 1]
+
+  alias Alerts.Alert
+  alias Alerts.InformedEntity
+  alias Alerts.Match
+  alias Dotcom.Alerts.Subway.Disruptions
+
+  @alerts_repo Application.compile_env!(:dotcom, :repo_modules)[:alerts]
+  @routes_repo Application.compile_env!(:dotcom, :repo_modules)[:routes]
+
+  @date_time Application.compile_env!(:dotcom, :date_time_module)
+
+  embed_templates "layouts/*"
+
+  def mount(_params, _session, socket) do
+    {:ok,
+     socket
+     |> assign(
+       :meta_description,
+       "Live service alerts for all MBTA transportation modes, including subway, bus, Commuter Rail, and ferry. " <>
+         "Updates on delays, construction, elevator outages, and more."
+     )
+     |> assign(:breadcrumbs, [Breadcrumb.build("Alerts")])
+     |> assign_now()
+     |> assign_route_type()
+     |> assign_routes()
+     |> assign(:subway_status, Dotcom.SystemStatus.subway_status())
+     |> assign(:disruption_groups, Disruptions.future_disruptions())
+     |> assign_alerts()
+     |> assign_banner_alert()
+     |> assign_alert_groups()}
+  end
+
+  def handle_params(%{"alerts_timeframe" => _} = params, uri, socket) do
+    new_query =
+      params
+      |> Map.delete("alerts_timeframe")
+      |> URI.encode_query()
+
+    new_path =
+      uri
+      |> URI.parse()
+      |> set_query(new_query)
+      |> to_path()
+
+    {:noreply, socket |> push_patch(to: new_path)}
```

Comment:
I'm kinda confused that you have to mess with encoding/parsing at all here, would something like this work?

```elixir
new_path = live_path(socket, __MODULE__, Map.drop(params, ["alerts_timeframe"]))
{:noreply, push_patch(socket, to: new_path)}
```

Reply:
[Ah yeah, that is much better!](https://github.com/mbta/dotcom/pull/2663/commits/15457054b7a992f81068b2bf9270ae7cffe7f9e6)

I tricked myself into thinking that `live_path` had been removed, because I couldn't figure out where it was defined. Looking again, I found it!